src/Module/RenderConfirmationBookingsHandler.php

```diff
@@ -0,0 +1,279 @@
+     *
+     * @var SelectCapableInterface
+     */
+    protected $servicesSelectRm;
+
+    /**
+     * The expression builder.
+     *
+     * @since [*next-version*]
+     *
+     * @var object
+     */
+    protected $exprBuilder;
+
+    /**
+     * The booking date and time format.
+     *
+     * @since [*next-version*]
+     *
+     * @var string|Stringable
+     */
+    protected $bookingFormat;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param TemplateInterface      $bookingTableTemplate The bookings table template.
+     * @param TemplateInterface      $bookingRowTemplate   The bookings table row template.
+     * @param SelectCapableInterface $bookingsSelectRm     The bookings SELECT resource model.
+     * @param SelectCapableInterface $servicesSelectRm     The services SELECT resource model.
+     * @param object                 $exprBuilder          The expression builder.
+     * @param string|Stringable      $bookingFormat        The bookings date and time format.
+     * @param string|Stringable|null $fallbackTz           The fallback timezone name.
+     */
+    public function __construct(
+        TemplateInterface $bookingTableTemplate,
+        TemplateInterface $bookingRowTemplate,
+        SelectCapableInterface $bookingsSelectRm,
+        SelectCapableInterface $servicesSelectRm,
+        $exprBuilder,
+        $bookingFormat,
+        $fallbackTz
+    ) {
+        $this->_setTemplate($bookingTableTemplate);
+        $this->_setFallbackTz($fallbackTz);
+        $this->bookingRowTemplate = $bookingRowTemplate;
+        $this->bookingsSelectRm   = $bookingsSelectRm;
+        $this->servicesSelectRm   = $servicesSelectRm;
+        $this->exprBuilder        = $exprBuilder;
+        $this->bookingFormat      = $bookingFormat;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function __invoke()
+    {
+        $event = func_get_arg(0);
+
+        if (!($event instanceof EventInterface)) {
+            throw $this->_createInvalidArgumentException(
+                $this->__('Argument is not an event instance'), null, null, $event
+            );
+        }
+
+        $payment   = $event->getParam(0);
+        $paymentId = $payment->ID;
+
+        // Alias expression builder
+        $b = $this->exprBuilder;
+
+        // Fetch the corresponding booking from storage
+        $condition = $b->eq($b->var('payment_id'), $b->lit($paymentId));
+        $bookings  = $this->bookingsSelectRm->select($condition);
+
+        $rows = '';
+        foreach ($bookings as $_booking) {
+            if (!($_booking instanceof BookingInterface)) {
+                throw $this->_createOutOfRangeException(
+                    $this->__('Booking is not a valid booking instance'), null, null, $_booking
+                );
+            }
+
+            $rows = $this->_renderBookingRow($_booking);
+        }
+
+        echo $this->_getTemplate()->render([
```

Comment:
It appears that there's no way to get the HTML of the table. I would very much prefer if there was a separate method for that.

Reply:
Didn't understand what you mean by not being able to get the HTML. This method returns the HTML, since it renders the template.

Reply:
This method *outputs* the HTML.

Reply:
Ah. Ok, I'll abstract.